trk/src/Screens/ClimbCreate/index.js

```diff
@@ -16,7 +16,30 @@ import { AuthContext } from '../../Utils/AuthContext';
 import SegmentedControl from '@react-native-segmented-control/segmented-control';
 
 
-const ClimbInputData = () => {
+const ClimbInputData = (props) => {
+
+
+
+  const { route } = props;
+  const climbData = route.params?.climbData;
+  const isEditMode = route.params?.editMode;
+
+  useEffect(() => {
+    if (isEditMode && climbData) {
+      // Set your state here based on climbData
+      setName(climbData.name);
+      setGrade(climbData.grade);
+      setGym(climbData.gym);
+      setType(climbData.type);
+      setSet(climbData.set);
+      setIfsc(climbData.ifsc);
+      setInfo(climbData.info);
+    }
+  }, [isEditMode, climbData]);
+
```

Comment:
Right, so this is where the tweaks take place. I'm basically checking whether your accessed the create form via home or profile. If the user is coming from profile (set detail's edit button), they will have a different UI.

Reply:
I call this path checking: EditMode.